### PR TITLE
R3SOL-368 Straighten dependencies

### PR DESCRIPTION
**Pull request by blsemo**
Remove unused dependencies and rejig a few classes to make the dependency graph of the ledger libs a lot flatter.

### CODE REVIEW

File: libs/ledger/ledger-utxo-data/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ dependencies {
 
     implementation project(':libs:crypto:crypto-core')
     implementation project(':libs:ledger:ledger-common-data')
-    implementation project(':libs:membership:membership-common')
+ //   implementation project(':libs:membership:membership-common')
```
Comment (williamvigorr3) [inline libs/ledger/ledger-utxo-data/build.gradle]: Should we delete this.

**Review by driessamyn — DISMISSED** [review]
LGTM

File: libs/ledger/ledger-utxo-data/build.gradle (hunk 1 above)
Comment (lankydan) [inline libs/ledger/ledger-utxo-data/build.gradle]: comment needs removing.

**Review by williamvigorr3 — COMMENTED** [review]
Looks good. Apart from the commented out dependency

**Review by lankydan — APPROVED** [review]

**Pull request merged**